rust/core-lib/src/rpc.rs

```diff
@@ -18,6 +18,8 @@
 //! the JSON-RPC protocol and the types here.
 //!
 //! [Serde]: https://serde.rs
+#![allow(clippy::needless_doctest_main)]
```

Comment:
this can also just be fixed, by removing the `# fn main() {` line from these various doctests; I think it was required at some point.

Reply:
Just remove it. Although I don't think this is a problem.

Reply:
it's not, we just prefer to minimize ignoring things. 🤷‍♂